src/check.h.in

```diff
@@ -659,6 +660,117 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
+/* OP may be any comparison operator. */
+#define _ck_assert_floating(X, OP, Y) do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  ck_assert_msg(_ck_x OP _ck_y, "Assertion '%s' failed: %s == %.6Lf, %s == %.6Lf", #X" "#OP" "#Y, #X, _ck_x, #Y, _ck_y); \
+} while (0)
+
+/**
+ * Check two floating point numbers to determine if X==Y
+ *
+ * If not X==Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_eq(X, Y) _ck_assert_floating(X, ==, Y)
+/**
+ * Check two floating point numbers to determine if X!=Y
+ *
+ * If not X!=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ne(X, Y) _ck_assert_floating(X, !=, Y)
+/**
+ * Check two floating point numbers to determine if X<Y
+ *
+ * If not X<Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_lt(X, Y) _ck_assert_floating(X, <, Y)
+/**
+ * Check two floating point numbers to determine if X<=Y
+ *
+ * If not X<=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_le(X, Y) _ck_assert_floating(X, <=, Y)
+/**
+ * Check two floating point numbers to determine if X>Y
+ *
+ * If not X>Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_gt(X, Y) _ck_assert_floating(X, >, Y)
+/**
+ * Check two floating point numbers to determine if X>=Y
+ *
+ * If not X>=Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_ge(X, Y) _ck_assert_floating(X, >=, Y)
+
+/**
+ * Check two floating point numbers to determine if X≈Y with specified precision
+ *
+ * If not X≈Y, the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param PREC positive number of decimal places to be compared
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.10.1
+ */
+#define ck_assert_floating_eq_prec(X, Y, PREC) \
+do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  unsigned char _prec = (PREC); \
```

Comment:
Thanks for the explanation. I now have a better appreciation for where you are trying to go with this macro.

Given your use of the macro, having this to help with math which is expected to be an approximate for fixed point seems reasonable. I still believe that for the floating point macros to be complete a more general macro which allow one to pass the threshold should be provided. Can you provide both in your change?